ci/jenkins/stop-stale-jobs.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+PULL_REQUEST_ID=""
+JENKINS_URL=""
+DEFAULT_WORKDIR=$HOME
+
+_usage="Usage: $0 [--pull-request <PullRequestId>] [--jenkins <JenkinsURL>]
+
+        --pull-request           Unique Pull Request id, to verify the stale jobs that need to be aborted.
+        --jenkins                Jenkins url to fetch the information about jobs."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --pull-request)
+    PULL_REQUEST_ID="$2"
+    shift 2
+    ;;
+    --jenkins)
+    JENKINS_URL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *) 
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function abort_queue_jobs {
+    echo "====== checking for capv stale waiting jobs on PR ${PULL_REQUEST_ID} ======"
+    queue="$(curl -s -XGET "${JENKINS_URL}queue/api/json" )"
+    length_of_queue="$(echo "$queue" | jq '[.[]][2]' | jq length)"
+
+    for (( j=0; j<$length_of_queue; j++ )); do
+            # Fetch the waiting job id and its pr id.
+            waiting_job_pr_id=$(echo "$queue" | jq -r "[.[]][2][$j].actions[1].parameters[9].value")
+            job_name=$(echo "$queue" | jq -r "[.[]][2][$j].task.name")
+
+            if [[ "$PULL_REQUEST_ID" == "$waiting_job_pr_id" ]]; then
+                echo "====== abort waiting job $job_name on PR ${PULL_REQUEST_ID} ======"
+                job_id=$(echo "$queue" | jq "[.[]][2][$j].id")
+                CURL_RETURN_CODE=0
+                curl -XPOST "${JENKINS_URL}/queue/cancelItem?id=$job_id" -u admin:$JENKINS_API_TOKEN || CURL_RETURN_CODE=$? || true
+                if [[ ${CURL_RETURN_CODE} -ne 0 ]]; then  
+                     echo "Curl connection failed with return code - ${CURL_RETURN_CODE}"
+                     exit 1
+                fi
+
+            fi
+    done
+}
+
+function abort_running_jobs {
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
```

Comment:
```suggestion
    echo "====== Checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
```

Please check and refine all comment sentences and capitalize the first char.

Reply:
@luolanzone , done, thanks!